auth/callback.go

```
@@ -0,0 +1,105 @@
+	if method == http.MethodGet {
+		query := url.Values{}
+		for key, value := range params {
+			query.Set(key, value)
+		}
+		req, err = http.NewRequestWithContext(ctx, method, urlStr+"?"+query.Encode(), nil)
+	} else {
+		data, err := json.Marshal(params)
+		if err != nil {
+			return err
+		}
+		req, err = http.NewRequestWithContext(ctx, method, urlStr, bytes.NewReader(data))
```

golangci-lint (1.62.2)

ineffectual assignment to err

(ineffassign)